src/app/data-planes/components/DataPlaneList.vue

```diff
@@ -23,10 +21,54 @@
     @change="emit('change', $event)"
   >
     <template #toolbar>
-      <slot name="toolbar" />
+      <FilterBar
+        class="data-plane-proxy-filter"
+        :placeholder="`tag: 'kuma.io/service: backend'`"
+        :query="props.query"
+        :fields="{
+          name: { description: 'filter by name or parts of a name' },
+          protocol: { description: 'filter by “kuma.io/protocol” value' },
+          service: { description: 'filter by “kuma.io/service” value' },
```

Comment:
I would guess that if you are viewing this from the ServiceDetail page then we don't want the `service` shortcut? (as its already being filtered by service)

Reply:
The more I think about it the more I think we should keep the toolbar slot and just pass slightly differently configured FilterBars and the KSelect into it. I know there is a bit of duplication, but its very simple code to duplicate, plus you save on the plumbing in DataPlaneList in adding the toolbar from The Outside

Reply:
I moved the toolbar back into the views. I’m generally not happy with the amount of duplicated markup for the sake of easier configurability. A lot of the times, it seems like a better idea to pass around data rather than markup.

Reply:
Cool, thanks. I agree that its not always super clear cut, and here it's definitely not.

Couple of points:

> A lot of the times, it seems like a better idea to pass around data rather than markup.

Markup _is_ data, just in a different format, its XML rather than JSON.

---

I think it might be worth making a `DataPlaneFilterBar` component which you would then slot into the toolbar slot in both places. As it stands right now tho, personally I'm still not totally sure I prefer the DataPlaneList component at all and we should just use AppCollection in both places. Just to be clear no action here, kinda talking out loud. I think once we get these description texts into i18n there'll be even less that could diverge.

I'm gonna take another proper look at the changes in a sec.